lib/options.py

```diff
@@ -213,6 +213,15 @@ def __init__(self):
                 Such files created by workers in the "var/reproduce" directory.
                 Note: The option works now only with parallel testing.""")
 
+        parser.add_argument(
+            "--server-start-timeout",
+            dest="server_start_timeout",
+            default=env_int('SERVER_START_TIMEOUT', 90),
+            type=int,
+            help="""Break the server process with kill signal if the server
+            starts longer than this amount of seconds. Default: 90 [seconds]."""
```

Comment:
An instance stopping process is a bit more complex than just sending SIGKILL.

Regular stopping of a default and a non-default instance is the following:

1. Send SIGTERM.
2. Start a watchdog, which will send SIGKILL after 5 seconds.
3. Wait termination of the process.
4. Stop the watchdog.

There is a process of stopping non-default servers, which remain alive after executing a test (disregarding whether it was succussful or failed). Here we use SIGKILL without any graceful period. See the `LuaTest.execute()` for advocation of this approach.

(If you think that it leads to problems I don't mind revisiting the approach if the problems it now solves will be taken into account. I guess the team will look at it in the scope of #318, but now this issue looks invalid or at least lacks necessary information.)

Back to the option description. I think that we should not clarify how exactly processes will be stopped, because it is beyond information, which is necessary to use the option.

Reply:
Also a server process can failed to start at all: so there is no process to send a signal.

Reply:
Important thing here is that we fail the test after this timeout if the server is unable to start (and if the server is not marked as expected to fail).